src/main/java/com/hazelcast/kubernetes/KubernetesConfig.java

```diff
@@ -0,0 +1,297 @@
+                            SERVICE_DNS.key(), SERVICE_NAME.key(), SERVICE_LABEL_NAME.key()));
+        }
+        if (serviceName != null && serviceLabelName != null) {
+            throw new InvalidConfigurationException(
+                    String.format("Properties '%s' and '%s' cannot be defined at the same time",
+                            SERVICE_NAME.key(), SERVICE_LABEL_NAME.key()));
+
+        }
+        if (serviceDnsTimeout < 0) {
+            throw new InvalidConfigurationException(
+                    String.format("Property '%s' cannot be a negative number", SERVICE_DNS_TIMEOUT.key()));
+        }
+        if (kubernetesApiRetries < 0) {
+            throw new InvalidConfigurationException(
+                    String.format("Property '%s' cannot be a negative number", KUBERNETES_API_RETIRES.key()));
+        }
+        if (servicePort < 0) {
+            throw new InvalidConfigurationException(
+                    String.format("Property '%s' cannot be a negative number", SERVICE_PORT.key()));
+        }
+    }
+
+    Mode getMode() {
+        if (serviceDns != null) {
+            return Mode.DNS_LOOKUP;
+        } else {
+            return Mode.KUBERNETES_API;
+        }
+    }
+
+    String getServiceDns() {
+        return serviceDns;
+    }
+
+    int getServiceDnsTimeout() {
+        return serviceDnsTimeout;
+    }
+
+    String getServiceName() {
+        return serviceName;
+    }
+
+    String getServiceLabelName() {
+        return serviceLabelName;
+    }
+
+    String getServiceLabelValue() {
+        return serviceLabelValue;
+    }
+
+    String getNamespace() {
+        return namespace;
+    }
+
+    boolean isResolveNotReadyAddresses() {
+        return resolveNotReadyAddresses;
+    }
+
+    int getKubernetesApiRetries() {
+        return kubernetesApiRetries;
+    }
+
+    String getKubernetesMasterUrl() {
+        return kubernetesMasterUrl;
+    }
+
+    String getKubernetesApiToken() {
+        return kubernetesApiToken;
+    }
+
+    String getKubernetesCaCertificate() {
+        return kubernetesCaCertificate;
+    }
+
+    int getServicePort() {
+        return servicePort;
+    }
+
+    @Override
+    public String toString() {
+        return "Kubernetes Discovery properties: { "
+                + "service-dns: " + serviceDns + ", "
+                + "service-dns-timeout: " + serviceDnsTimeout + ", "
+                + "service-name: " + serviceName + ", "
+                + "service-port: " + servicePort + ", "
+                + "service-label: " + serviceLabelName + ", "
+                + "service-label-value: " + serviceLabelValue + ", "
+                + "namespace: " + namespace + ", "
+                + "resolve-not-ready-addresses: " + resolveNotReadyAddresses + ", "
+                + "kubernetes-api-retries: " + kubernetesApiRetries + ", "
+                + "kubernetes-master: " + kubernetesMasterUrl + "}";
+    }
+
+    enum Mode {
```

Comment:
DiscoveryMode can be more descriptive here

Reply:
Changed. Thanks!